src/common/guards/api-key/api-key.guard.ts

```diff
@@ -0,0 +1,20 @@
+import { CanActivate, ExecutionContext, Injectable, UnauthorizedException } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { Observable } from 'rxjs';
+
+const configService = new ConfigService();
+@Injectable()
+export class ApiKeyGuard implements CanActivate {
+  canActivate(context: ExecutionContext): boolean | Promise<boolean> | Observable<boolean> {
+    const apiKey = configService.getOrThrow<string>('SERVER_API_KEY');
+    const request = context.switchToHttp().getRequest();
+
+    const authHeader = request.headers['authorization'];
+
+    if (authHeader && authHeader === `Bearer ${apiKey}`) {
+      return true;
+    }
+
+    throw new UnauthorizedException('Invalid API key');
```

Comment:
Are our messages not coming from a common file?

Reply:
No, we did not have any cusstom error messages so far.
Recent ones we added for...

throw new BadRequestException(`Channel ${notification.channelType} is not enabled`);
 throw new UnauthorizedException('Invalid API key');